File: sortInWave.scala
object SortInWaves extends App {
  val arr = Array(10, 90, 49, 2, 1, 5, 23)
                //90 ,10 ,49 ,1 ,5 ,2 ,23 
  // Traverse all even indices and print the elements at even indices
  SortInWav(arr)
  arr.foreach(println)
}

def SortInWav(arr:Array[Int])={
   for (i <- 0 until arr.length-1 by 2) {
    if(i>0 && arr(i)< arr(i-1)){ 
        val temp=arr(i)
        arr(i)=arr(i-1)
        arr(i-1)=temp
    }
    if(i<arr.length-1 && arr(i)< arr(i+1)){ 
        val temp=arr(i)
        arr(i)=arr(i+1)
        arr(i+1)=temp
    }
  }
}

File: SortInWave/sortInWave.py

from functools import reduce
def sortInWave(acc, i):

    # Traverse all even elements
        arr=acc
        # If current even element is smaller than previous
        if (i > 0 and arr[i] < arr[i-1]):
            arr[i], arr[i-1] = arr[i-1], arr[i]

        # If current even element is smaller than next
        if (i < len(arr)-1 and arr[i] < arr[i+1]):
            arr[i], arr[i+1] = arr[i+1], arr[i]

        return arr

arr = [10, 90, 49, 2, 1, 5, 23]
reduce(sortInWave,range(0,len(arr),2),arr)
for i in range(0, len(arr)):
    print(arr[i])